src/transformers/models/bloom/tokenization_bloom_fast.py

```diff
@@ -155,6 +165,81 @@ def _encode_plus(self, *args, **kwargs) -> BatchEncoding:
 
         return super()._encode_plus(*args, **kwargs)
 
+    def batch_decode(
+        self,
+        sequences: Union[List[int], List[List[int]], "np.ndarray", "torch.Tensor", "tf.Tensor"],
+        skip_special_tokens: bool = False,
+        clean_up_tokenization_spaces: bool = True,
+        **kwargs
+    ) -> List[str]:
+        """
+        Convert a list of lists of token ids into a list of strings by calling decode.
+
+        Args:
+            sequences (`Union[List[int], List[List[int]], np.ndarray, torch.Tensor, tf.Tensor]`):
+                List of tokenized input ids. Can be obtained using the `__call__` method.
+            skip_special_tokens (`bool`, *optional*, defaults to `False`):
+                Whether or not to remove special tokens in the decoding.
+            clean_up_tokenization_spaces (`bool`, *optional*, defaults to `True`):
+                Whether or not to clean up the tokenization spaces.
+            kwargs (additional keyword arguments, *optional*):
+                Will be passed to the underlying model specific decode method.
+
+        Returns:
+            `List[str]`: The list of decoded sentences.
+        """
+        if clean_up_tokenization_spaces:
+            logger.warning(
+                "Bloom tokenizer was built in order to have lossless encoding. Most likely, you should use"
+                " `clean_up_tokenization_spaces=False`. This flag is deprecated and will be remove in v5. The"
+                " behaviour will correspond to `clean_up_tokenization_spaces=False`"
+            )
```

Comment:
```suggestion
            warnings.warn(
                "Bloom tokenizer was built in order to have lossless encoding. Most likely, you should use"
                " `clean_up_tokenization_spaces=False`. This flag is deprecated and will be remove in v5. The"
                " behaviour will correspond to `clean_up_tokenization_spaces=False`"
                FutureWarning,
            )
```
Let's use a `FuturWarning` here 😉

Reply:
Any reason to use `warnings.warn` instead of `logger.warning`?

Reply:
FuturWarning will be easier to catch once we do a cleanup and are bound to disappear, while other warnings might not

Reply:
What I meant is can't you use `logger.warning` and have a FutureWarning? Well I just searched in the repo and all seem to be using a `warning.warn` so let's do it your way!